api/app/src/main/kotlin/packit/service/OneTimeTokenService.kt

```diff
@@ -0,0 +1,52 @@
+package packit.service
+
+import jakarta.persistence.EntityManager
+import jakarta.transaction.Transactional
+import org.springframework.http.HttpStatus
+import org.springframework.stereotype.Service
+import packit.exceptions.PackitException
+import packit.model.OneTimeToken
+import packit.model.Packet
+import packit.repository.OneTimeTokenRepository
+import java.time.Instant
+import java.util.*
+
+interface OneTimeTokenService
+{
+    fun createToken(packetId: String, filePaths: List<String>): OneTimeToken
+    fun getToken(id: UUID): OneTimeToken
+    fun cleanUpExpiredTokens()
+}
+
+private const val EXPIRY_BUFFER_SECONDS = 30L
+
+@Service
+class BaseOneTimeTokenService(
+    private val oneTimeTokenRepository: OneTimeTokenRepository,
+    private val entityManager: EntityManager
+) : OneTimeTokenService {
+    @Transactional
+    override fun createToken(packetId: String, filePaths: List<String>): OneTimeToken {
+        val packet = entityManager.getReference(Packet::class.java, packetId)
```

Comment:
You should inject abstract away from entityManager and call `PacketService` to get packets here unless you have to for some reason.